### PR TITLE
fix: switch to golangci-lint and fix all the lint

### PR DESCRIPTION
**Pull request by FUSAKLA**
Ugh.. But all the tests are passing! :D

### Motivation
Historically we used just the revive tool and there was a lot of warning not solved anyway.

So I decided to switch to the de facto standard for Go linting golangci-lint (which embeds also revive)

This PR switches the linter AND fixes all the lint

### CODE REVIEW

**Comment by FUSAKLA** [comment]
tried all the examples locally, everything seems to be working

e2e tests are passing

File: pkg/storage/container_test.go [hunk 1]
```diff
@@ -22,7 +23,7 @@ func Test_Container_Add(t *testing.T) {
 			NewInMemoryCappedContainer(containerCapacity),
 		}
 		for _, c := range containers {
-			t.Run(fmt.Sprintf(" | %s | %s", reflect.TypeOf(c).String(), tt.name), func(t *testing.T) {
+			t.Run(fmt.Sprintf(" | %s | %s", reflect.TypeOf(c).String(), tt.name), func(_ *testing.T) {
```
Comment (rudo-thomas) [inline pkg/storage/container_test.go:26]: Why `t.Run` when the testing.T on the inside is not used?

File: pkg/tailer/tailer_test.go [hunk 2]
```diff
@@ -293,10 +304,15 @@ func offsetPersistenceTestRun(t offsetPersistenceTest) error {
 	defer f.Close()
 
 	eventCount := make(chan int)
-	persistPositionInterval, _ := time.ParseDuration("10s")
+	persistPositionInterval, err := time.ParseDuration("10s")
+	if err != nil {
+		return fmt.Errorf("Error while parsing duration: %w", err)
```
Comment (rudo-thomas) [inline pkg/tailer/tailer_test.go:309]:
```suggestion
		return fmt.Errorf("error while parsing duration: %w", err)
```

File: pkg/storage/container_test.go (hunk 1 above)
Comment (FUSAKLA) [reply pkg/storage/container_test.go:26]: It's easier to distinguish which test case failed of the whole test run by its name

File: pkg/kafka_ingester/kafka_ingester.go [hunk 3]
```diff
@@ -181,13 +182,10 @@ func (k *KafkaIngester) Run() {
...
 			k.done = true
 		}()
-		for {
-			select {
-			case <-k.shutdownChannel:
-				cancel()
-				k.kafkaReader.Close()
-				return
-			}
+		for range k.shutdownChannel {
```
[the remark this answers is not in the document]
Comment (FUSAKLA) [reply pkg/kafka_ingester/kafka_ingester.go:185]: Right, lemme fix it

File: pkg/envoy_access_log_server/service_v3_test.go [hunk 4]
```diff
@@ -237,7 +236,7 @@ func Test_exportHTTPResponsePropertiesV3(t *testing.T) {
 // full extent extraction of request, response and common properties are to be tested separately within:
 // Test_exportHTTPResponsePropertiesV3
 // Test_exportHTTPRequestPropertiesV3
-// Test_exportCommonPropertiesV3
+// Test_exportCommonPropertiesV3.
```
[the remark this answers is not in the document]
Comment (FUSAKLA) [reply pkg/envoy_access_log_server/service_v3_test.go:239]: nah, lets drop the godot

File: .golangci.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      - io/ioutil.ReadFile
+      - io.Copy(*bytes.Buffer)
+      - io.Copy(os.Stdout)
+
+  govet:
+    disable:
+      - fieldalignment # I'm ok to waste some bytes
+
+  nakedret:
+    # No naked returns, ever.
```
[the remark this answers is not in the document]
Comment (FUSAKLA) [reply .golangci.yaml:67]: dropping the comment, the config is taken from... somewhere but I cannot find the source anymore (It's been awhile) :confounded:

File: pkg/envoy_access_log_server/service_v3_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (FUSAKLA) [reply pkg/envoy_access_log_server/service_v3_test.go]: removed godot here https://github.com/seznam/slo-exporter/pull/117/commits/1d3cc860b7fe8c77db364d7f1263fecc2326cd02

File: pkg/kafka_ingester/kafka_ingester.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (FUSAKLA) [reply pkg/kafka_ingester/kafka_ingester.go]: PTAL https://github.com/seznam/slo-exporter/pull/117/commits/1d3cc860b7fe8c77db364d7f1263fecc2326cd02

File: .golangci.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (FUSAKLA) [reply .golangci.yaml]: https://github.com/seznam/slo-exporter/pull/117/commits/1d3cc860b7fe8c77db364d7f1263fecc2326cd02

**Review by rudo-thomas — APPROVED** [review]

**Pull request merged**